src/commands.ts

```diff
@@ -18,5 +18,10 @@ export const commands = async (message: Message) => {
     const matching = commandList
         .find(({ triggers }) => triggers
             .find((trigger) => trigger === command)) || { command: fallback };
-    message.channel.send(await matching.command(args.slice(command.length + 1), embed, message));
+
+    if (matching.command === fallback) {
+        message.channel.send(await matching.command(command, embed, message));
```

Comment:
@BOLT04 I wrote the code like this just to make the command suggestion start working first. I too love the look of a clean `index.ts`. I would definitely consider abstracting the conditional away like this:
* each `commandHandler` would get the full command passed in `arg` with the command name too
* it would be the job of each `commandHandler` to parse out the actual arguments if needed and use them (I think `bio` does it already)
* `fallback`, just like the current version of `help` doesn't have to bother doing any parsing really but here's a suggestion for `help` that just occurred to me - `help` should have the option of taking in command names and display the info about the specific command

I'd like to hear thoughts on this sort of a refactor. :slightly_smiling_face:

Reply:
@RaisinTen thank you for explaining and giving more suggestions 😃. I also prefer iterative development, so exactly what you did. First focus on getting it to work, then we can refactor 😄.

I agree with all of your suggestions 👍. Here are my suggestions:
> each commandHandler would get the full command passed in arg with the command name too

We can change the interface of a `commandHandler` so that the first parameter is a [tuple](https://www.typescriptlang.org/docs/handbook/basic-types.html#tuple)) where the first value is the command name, and the second is the full command passed: `arg: [string, string]`.

What do you think of opening 2 new issues, one for the refactor of the `commandHandler` interface, another for the new feature on `help` to receive arguments and display info as you said @RaisinTen ? We can merge this and separate those improvements for later 👍

Reply:
@BOLT04 yes, that's exactly what I was aiming for! :smile:
Sure, I'll open the issues. :slightly_smiling_face:

Reply:
@BOLT04 done! :slightly_smiling_face: